library/Icinga/Web/Navigation/ConfigMenu.php

```diff
@@ -0,0 +1,293 @@
+    {
+        $cogMenuItem->add([
+            HtmlElement::create(
+                'a',
+                [],
+                [
+                    new Icon('cog'),
+                    $this->createHealthBadge(),
+                ]
+            ),
+            $this->createLevel2Menu()
+        ]);
+
+        if ($this->cogItemActive) {
+            $cogMenuItem->addAttributes([ 'class' => 'active' ]);
+        }
+    }
+
+    protected function assembleLevel2Nav(BaseHtmlElement $level2Nav)
+    {
+        $navContent = HtmlElement::create('div', ['class' => 'flyout-content']);
+        foreach ($this->children as $c) {
+            if (isset($c['permission']) && ! Auth::getInstance()->hasPermission($c['permission'])) {
+                continue;
+            }
+
+            if (isset($c['title'])) {
+                $navContent->add(HtmlElement::create(
+                    'h3',
+                    [],
+                    t($c['title'])
+                ));
+            }
+
+            $ul = HtmlElement::create('ul', ['class' => 'nav']);
+            foreach ($c['items'] as $key => $item) {
+                $ul->add($this->createLevel2MenuItem($item, $key));
+            }
+            $navContent->add($ul);
+        }
+        $level2Nav->add($navContent);
+    }
+
+    protected function getHealthCount()
+    {
+        $count = 0;
+        $title = null;
+        $worstState = null;
+        foreach (HealthHook::collectHealthData()->select() as $result) {
+            if ($worstState === null || $result->state > $worstState) {
+                $worstState = $result->state;
+                $title = $result->message;
+                $count = 1;
+            } elseif ($worstState === $result->state) {
+                $count++;
+            }
+        }
+
+        switch ($worstState) {
+            case HealthHook::STATE_OK:
+                $count = 0;
+                break;
+            case HealthHook::STATE_WARNING:
+                $this->state = self::STATE_WARNING;
+                break;
+            case HealthHook::STATE_CRITICAL:
+                $this->state = self::STATE_CRITICAL;
+                break;
+            case HealthHook::STATE_UNKNOWN:
+                $this->state = self::STATE_UNKNOWN;
+                break;
+        }
+
+        $this->title = $title;
+
+        return $count;
+    }
+
+    protected function isSelectedItem($item)
+    {
+        if ($item !== null && Icinga::app()->getRequest()->getUrl()->matches($item['url'])) {
+            $this->selected = $item;
+            return true;
+        }
+
+        return false;
+    }
+
+    protected function createHealthBadge()
+    {
+        return $this->getHealthCount() == 0 ? '' : new StateBadge($this->getHealthCount(), 'critical');
+    }
+
+    protected function createLevel2Menu()
+    {
+        $level2Nav = HtmlElement::create(
+            'div',
+            Attributes::create(['class' => 'nav-level-2 flyout'])
```

Comment:
If you already use `Attributes::create`, please don't use `HtmlElement::create` but the constructor `new HtmlElement()`.